sim/team-validator.ts

```diff
@@ -1767,6 +1775,12 @@ export class TeamValidator {
 		}
 		// Event-related ability restrictions only matter if we care about illegal abilities
 		const ruleTable = this.ruleTable;
+		if (ruleTable.has('obtainablemoves')) {
+			if (eventData.generation > setSources.maxSourceGen() && !(dex.gen <= 2 && eventSpecies.gen === 1)) {
+				if (fastReturn) return true;
+				problems.push(`${name} must not have moves only learnable in gen ${setSource.maxSourceGen()}${etc}.`);
```

Comment:
Would appreciate suggestions for this message

Reply:
Maybe just ``${name} does not learn ${move.name} in gen ${dex.gen}${etc}.``